flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -109,22 +113,22 @@
 
 	private RestartStrategies.RestartStrategyConfiguration restartStrategyConfiguration;
 	
-	private long taskCancellationIntervalMillis = ConfigConstants.DEFAULT_TASK_CANCELLATION_INTERVAL_MILLIS;
+	private long taskCancellationIntervalMillis = -1;
 
 	// Serializers and types registered with Kryo and the PojoSerializer
 	// we store them in linked maps/sets to ensure they are registered in order in all kryo instances.
 
-	private final LinkedHashMap<Class<?>, SerializableSerializer<?>> registeredTypesWithKryoSerializers = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, SerializableSerializer<?>>> registeredTypesWithKryoSerializers = new UserCodeValue<>(new LinkedHashMap<Class<?>, SerializableSerializer<?>>());
 
-	private final LinkedHashMap<Class<?>, Class<? extends Serializer<?>>> registeredTypesWithKryoSerializerClasses = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>> registeredTypesWithKryoSerializerClasses = new UserCodeValue<>(new LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>());
 
-	private final LinkedHashMap<Class<?>, SerializableSerializer<?>> defaultKryoSerializers = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, SerializableSerializer<?>>> defaultKryoSerializers = new UserCodeValue<>(new LinkedHashMap<Class<?>, SerializableSerializer<?>>());
 
-	private final LinkedHashMap<Class<?>, Class<? extends Serializer<?>>> defaultKryoSerializerClasses = new LinkedHashMap<>();
+	private final UserCodeValue<LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>> defaultKryoSerializerClasses = new UserCodeValue<>(new LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>());
 
-	private final LinkedHashSet<Class<?>> registeredKryoTypes = new LinkedHashSet<>();
+	private final UserCodeValue<LinkedHashSet<Class<?>>> registeredKryoTypes = new UserCodeValue<>(new LinkedHashSet<Class<?>>());
 
-	private final LinkedHashSet<Class<?>> registeredPojoTypes = new LinkedHashSet<>();
+	private final UserCodeValue<LinkedHashSet<Class<?>>> registeredPojoTypes = new UserCodeValue<>(new LinkedHashSet<Class<?>>());
```

Comment:
Thanks for the fix.

The original motivation for adding it was the following:
Users requested a way to have one central configuration file for all their job's UDFs. With setting global job parameters (for example the ParameterTool), they can use a properties file and the KVs from there are accessible from all UDFs.
Also, the same user wanted to have these configuration properties accessible in the webinterface. So if the `toMap()` method is overwritten (the ParameterTool is doing that), the job parameters are accessible from the web interface.